migrations/2024-06-14-081653_add-lib-and-bins-columns/up.sql

```diff
@@ -0,0 +1,6 @@
+alter table versions
+    add has_lib boolean,
+    add bin_names text[];
+
+comment on column versions.has_lib is 'TRUE if the version has a library (e.g. `src/lib.rs`), FALSE if no library was detected, or NULL if the version has not been analyzed yet.';
+comment on column versions.bin_names is 'list of the names of all detected binaries in the version. the list may be empty which indicates that no binaries were detected in the version. the column may be NULL is the version has not been analyzed yet.';
```

Comment:
IIUC, null is used for existing versions that need backfilling? And for newly published versions, a non-null value would be expected to be inserted? Do we also plan to alter the table to non-nullable once all values are filled?

Reply:
> IIUC, null is used for existing versions that need backfilling? And for newly published versions, a non-null value would be expected to be inserted?

correct

> Do we also plan to alter the table to non-nullable once all values are filled?

I'm not sure yet. at the moment the detection is done inside the publish endpoint, but if we were to move the detection to an async background worker job we might want to keep this "no-yet-analyzed" state.

Reply:
Thanks for the explanation!

This actually reminds me of `default_versions`, which faces a similar situation but actually handles it quite well.